### PR TITLE
Added new tests for contract in negative scenarios

### PR DESCRIPTION
**Pull request by darthsiroftardis**
# Description
Added new test cases for contract in negative scenarios

### CODE REVIEW

File: tests/src/lib.rs [hunk 1]
```diff
@@ -26,6 +26,15 @@ mod tests {
...
     }
 
+    #[test]
+    #[should_panic]
+    fn test_erc20_transfer_too_much() {
+        let amount = ERC20_INIT_BALANCE + 1;
+        let mut token = ERC20Contract::deployed();
+        token.transfer(BOB,amount,Sender(ALI));
+    }
+
```
Comment (zie1ony) [inline tests/src/lib.rs:36]: Too many lines.

File: tests/src/lib.rs [hunk 2]
```diff
@@ -26,6 +26,15 @@ mod tests {
         assert_eq!(token.balance_of(BOB), amount);
     }
 
+    #[test]
+    #[should_panic]
+    fn test_erc20_transfer_too_much() {
+        let amount = ERC20_INIT_BALANCE + 1;
+        let mut token = ERC20Contract::deployed();
+        token.transfer(BOB,amount,Sender(ALI));
```
Comment (zie1ony) [inline tests/src/lib.rs:34]: Fix comas everywhere. There should be a whitespace between args.

File: tests/src/lib.rs [hunk 3]
```diff
@@ -37,6 +46,17 @@ mod tests {
...
+    #[should_panic]
+    fn test_approve_too_much() {
+        let allowance = 1;
+        let amount = 2;
+        let  mut token = ERC20Contract::deployed();
+        token.approve(BOB,allowance,Sender(ALI));
+        token.transfer_from(ALI,JOE,amount,Sender(BOB));
+    }
+
+
```
Comment (zie1ony) [inline tests/src/lib.rs:59]: Empty line.

File: tests/src/lib.rs [hunk 4]
```diff
@@ -37,6 +46,17 @@ mod tests {
         assert_eq!(token.allowance(BOB, ALI), 0);
     }
 
+    #[test]
+    #[should_panic]
+    fn test_approve_too_much() {
```
Comment (zie1ony) [inline tests/src/lib.rs:51]: This test fails at `token.transfer_from` not at `token.approve`. You can remove this test, as approve never panics.

File: tests/src/lib.rs [hunk 5]
```diff
@@ -49,4 +69,12 @@ mod tests {
         assert_eq!(token.balance_of(JOE), amount);
         assert_eq!(token.allowance(ALI, BOB), allowance - amount);
     }
+
+    #[test]
+    #[should_panic]
+    fn test_erc20_transfer_from_too_much() {
+        let amount = ERC20_INIT_BALANCE + 1;
```
Comment (zie1ony) [inline tests/src/lib.rs:76]: The amount can be `1` as well.

Comment (darthsiroftardis) [reply tests/src/lib.rs:76]: The amount that initialized within the contract is the INIT balance, If the amount is 1, then the transfer actual goes through.

Comment (zie1ony) [reply tests/src/lib.rs:64]: No it's not.

Comment (darthsiroftardis) [reply tests/src/lib.rs:64]: There was an error in the test, I wrote. Fixed it.

**Pull request merged**